editor/src/components/canvas/controls/select-mode/select-mode.spec.browser2.tsx

```diff
@@ -651,6 +651,99 @@ describe('Select Mode Double Clicking With Fragments', () => {
   })
 })
 
+describe('Select Mode Double Clicking With conditionals', () => {
+  it('Double click can dive into single conditional inside element with an expression in the active branch', async () => {
+    // prettier-ignore
+    const desiredPaths = createConsecutivePaths(
+      'sb' +                // Skipped as it's the storyboard
+      '/scene-2' +          // Skipped because we skip over Scenes
+      '/Card-instance' +    // Skipped because we skip component children of Scenes
+      ':Card-Root' +        // Skipped because we skip over root elements
+      '/Card-Row-Buttons',  // <- Single click
+      '/Card-Button-3',     // <- Double click
+    )
+
+    const renderResult = await renderTestEditorWithCode(
+      TestProjectAlpineClimbWithConditional,
+      'await-first-dom-report',
+    )
+
+    const cardSceneRoot = renderResult.renderedDOM.getByTestId('card-scene')
+    const cardSceneRootBounds = cardSceneRoot.getBoundingClientRect()
+
+    const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    const doubleClick = createDoubleClicker(
+      canvasControlsLayer,
+      cardSceneRootBounds.left + 130,
+      cardSceneRootBounds.top + 220,
+    )
+
+    await fireSingleClickEvents(
+      canvasControlsLayer,
+      cardSceneRootBounds.left + 130,
+      cardSceneRootBounds.top + 220,
+    )
+
+    checkFocusedPath(renderResult, null)
+    checkSelectedPaths(renderResult, [desiredPaths[0]])
+
+    await doubleClick()
+    checkFocusedPath(renderResult, null)
+    checkSelectedPaths(renderResult, [desiredPaths[1]])
+
+    await doubleClick()
+    checkFocusedPath(renderResult, null)
+    // can not dive deeper into the conditional because it has a sibling
+    checkSelectedPaths(renderResult, [desiredPaths[1]])
```

Comment:
something is fishy here. the desiredPaths in these tests should not contain paths you cannot reach!

Reply:
actually all these tests are missing an assertion that the desiredPaths were eventually exhausted

Reply:
I doesn't contain paths we can not reach. I just explicitly made another double click with the same assertion to show that another double click doesn't change the selected element.

Reply:
I'm not sure what you mean, I think this test proves that the desiredPaths is exhausted.

Reply:
ah ok, so desiredPaths.length in this case is 2?